src/main/scala/org/labrad/Server.scala

```diff
@@ -262,13 +264,58 @@ object Server {
    * Includes basic parsing of command line options.
    */
   def run(s: Server[_, _], args: Array[String]): Unit = {
-    val options = Util.parseArgs(args, Seq("name", "host", "port", "password"))
+    val parser = new ArgotParser("labrad",
```

Comment:
Do you mean building the parser in a separate method? The tricky thing is that adding each parameter creates a separate parameter object, so the `defineParameters` method would have to return the parser as well as all those individual parameter objects. I'm not sure how much of a win it would be.